Magento_Theme/web/js/view/movies/movie-modal.js

```diff
@@ -0,0 +1,114 @@
+define(['uiElement', 'jquery'], (uiElement, $) => {
+  return uiElement.extend({
+    defaults: {
+      template: 'Magento_Theme/movies/movie-modal-layout',
+      links: {
+        movieList: '${ $.parentName }:movieList',
+        movie: '${ $.parentName }:movie',
+        modalMode: '${ $.parentName }:modalMode',
+      },
+      tracks: {
+        movieList: true,
+        movie: true,
+      },
+      modules: {
+        parentComponent: '${ $.parentName }',
+      },
+      listens: {
+        movie: 'handleMovieInputFields',
+      },
+    },
+
+    initialize() {
+      this._super();
+    },
+
+    randomIdNumber() {
+      const randomNumber = Math.floor(Math.random() * 100000).toString();
+
+      return 'tt00' + randomNumber.padStart(5, '0');
+    },
+
+    clearInputFields() {
+      $('#title').val('');
+      $('#year').val('');
+      $('#poster').val('');
+    },
+
+    closeMovieModal(e) {
+      e.preventDefault();
+      $('.movie-modal').hide();
+    },
+
+    handleMovieInputFields() {
+      setTimeout(() => {
```

Comment:
why you need setTimeout here? use events
https://developer.adobe.com/commerce/frontend-core/javascript/jquery-widgets/modal/#events